Tests/Add-ZipArchiveEntry.Tests.ps1

```diff
@@ -0,0 +1,273 @@
+
+#Requires -Version 5.1
+Set-StrictMode -Version 'Latest'
+
+Import-Module -Name (Join-Path -Path $PSScriptRoot -ChildPath '..\Zip' -Resolve) -Force
+Import-Module -Name (Join-Path -Path $PSScriptRoot -ChildPath '..\PSModules\Glob' -Resolve) -Force
+
+$archive = $null
+
+function GivenFile
+{
+    param(
+        [string[]]
+        $Path,
+
+        $Content
+    )
+
+    foreach( $pathItem in $Path )
+    {
+        $fullPath = Join-Path -Path $TestDrive.FullName -ChildPath $pathItem
+
+        $parentDir = $fullPath | Split-Path
+        if( -not (Test-Path -Path $parentDir -PathType Container) )
+        {
+            New-Item -Path $parentDir -ItemType 'Directory'
+        }
+
+        if( -not (Test-Path -Path $fullPath -PathType Leaf) )
+        {
+            New-Item -Path $fullPath -ItemType 'File'
+        }
+
+        if( $Content )
+        {
+            [IO.File]::WriteAllText($fullPath,$Content)

```

Comment:
Could you let me know what the reason/edge-case is for using `[IO.File]::WriteAllText` over `Set-Content`?

Reply:
`Get-Content`/`Set-Content` add newlines.